sources/platform/workflow/job/overview.md

```diff
@@ -114,6 +119,59 @@ When triggered, a Job does the following:
+
+    Example:
+
+    In the above assembly line, `job-2` and `job-3` are triggered manually. Suppose, `job-2` completes first then it will trigger `job-5` once but not `job-4` since it has an inconsistent IN `job-1`. As `job-3` is still in processing state, `job-5` will remain in queued state until `job-3` is complete. Once `job-3` completes it will trigger `job-5` again. `job-5` will have 2 queued builds for it. If `job-7` is in processing state, then `job-5` will wait for that also to complete and then both of its builds will go to processing state one by one.
+
+* **immediate dependencyMode**
+
+    As the name suggests, jobs are triggered immediately without waiting for any incomplete IN/OUT jobs to be completed and jobs also get triggered even if IN resources are inconsistent. When a job completes or a resource is updated, all the consistent jobs which are using this resource/job as IN are triggered.
+
+    Example:
+
+    In the above assembly line, `job-2` and `job-3` are triggered manually. Suppose, `job-2` completes first then it will trigger `job-5` and `job-4` both even though `job-4` has an inconsistent IN `job-1`. While `job-3` is still in processing state, `job-5`and `job-4` will goto processing state immediately if required minions(or nodes) are available even if `job-7` and `job-6` are processing. Now, when job `job-3` completes, it will trigger
+    `job-5` again and this build will go to processing state as soon as first build for `job-5` is complete.
+
+* **strict dependencyMode**
+
+    In the `chrono` and `immediate` modes, dependent jobs are triggered multiple times when more than two IN jobs are running. In `strict` mode we only trigger a job once when more than one IN jobs completes. When a job completes or a resource is updated, all the consistent jobs which are using this resource/job as IN are evaluated and the ones which have consistent(valid yml definition and success state) and no processing IN resources/jobs are triggered, `waiting jobs` are created for consistent dependencies which have processing IN jobs.
+
+    You can see the waiting jobs on dashboards' grid-view. In case your waiting job is not getting processed and all the IN jobs are in complete state then you can delete them.
+
+    <img src="/images/platform/jobs/waiting-jobs.png" alt="How are DevOps activities triggered?" style="width:100%;vertical-align: middle;display: block;margin-left: auto;margin-right: auto;"/>
+
+    Example:
+
+    In the above assembly line, `job-2` and `job-3` are triggered manually. Suppose, `job-2` completes first then it will not trigger any job. It will not trigger `job-5` since `job-3` is still running and it will not trigger `job-4` since it has an inconsistent IN `job-1`. A `waiting job` is created for `job-5`. Once `job-3` completes it will trigger `job-5`. `job-5` will then go to processing state even if job `job-7` is processing.
```

Comment:
In the above assembly line, `job-2` and `job-3` are triggered manually. Suppose `job-2` completes first.  In this case, it will not trigger any subsequent job. It will not trigger `job-5` since `job-3` is still running and it will not trigger `job-4` since it has an inconsistent IN `job-1`. Instead, a `waiting job` is created for `job-5`. Once `job-3` completes it will trigger `job-5`. `job-5` will then be removed from the `waiting jobs` list and move to a queued state even if `job-7` is processing.  Once in a queued state, the job will be picked up to run as soon as minions are available.